modules/pam_motd/pam_motd.c

```diff
@@ -166,9 +166,24 @@ static int compare_strings(const void *a, const void *b)
     }
 }
 
+static const char* motddir;
+
 static int filter_dirents(const struct dirent *d)
 {
-    return (d->d_type == DT_REG || d->d_type == DT_LNK);
+    struct stat s;
+    char *fullpath;
+    int rc;
+
+    while(1){
```

Comment:
There is no loop here, why `while`?

Reply:
I'm "grouping" the conditions inside, enables me to use break.

Reply:
The loop is not needed, you can just do `return 0` instead;

Reply:
Yup agree, just a personal "preference".